src/ucs/datastruct/piecewise_func.c

```diff
@@ -41,38 +55,51 @@ void ucs_piecewise_func_cleanup(ucs_piecewise_func_t *func)
     }
 }
 
-static void
-ucs_piecewise_func_assert_covered_range(const ucs_piecewise_func_t *func)
+static void ucs_piecewise_func_check(ucs_piecewise_func_t *func)
 {
 #if ENABLE_ASSERT
     ucs_piecewise_segment_t *tail = ucs_list_tail(&func->head,
                                                   ucs_piecewise_segment_t,
                                                   list);
+    ucs_piecewise_segment_t *head = ucs_list_head(&func->head,
```

Comment:
get head/tail only after checking list is not empty

Reply:
Done.